### PR TITLE
Initial commit of support for the new capability descriptors

### PR DESCRIPTION
**Pull request by autodidaddict**

### CODE REVIEW

File: src/capability.rs [hunk 1]
```diff
@@ -83,22 +73,40 @@ impl NativeCapability {
...
     }
+
+    /// Returns the full descriptor for the capability provider
+    pub fn descriptor(&self) -> &CapabilityDescriptor {
+        &self.descriptor
+    }
+}
+
+fn get_descriptor(plugin: &Box<dyn CapabilityProvider>) -> Result<CapabilityDescriptor> {
+    let res = plugin.handle_call("system", OP_GET_CAPABILITY_DESCRIPTOR, &[])?;
```
Comment (ewbankkit) [inline src/capability.rs:109]: As I recall there was talk at one time of making `system` a constant defined somewhere.
Would now be a time to do that?

**Comment by autodidaddict** [comment]
Sadly the best time was when making the change in the codec. We should
probably capture that as an issue on the codec crate so it doesn't slip
through the cracks again

File: src/capability.rs (hunk 1 above)
Comment (ewbankkit) [reply src/capability.rs:109]: Added https://github.com/wascc/wascc-codec/issues/13 to capture this.

**Comment by ewbankkit** [comment]
LGTM.

**Pull request merged**